File: problem-2/main.js
function fib(n){
    if (n == 1 || n == 0) {
        return 1
    }
    else {
        return fib(n-1) + fib(n-2)
    }
}
var sum = 0
var i = 0
while (true) {
    x = fib(i)
    if (x > 4_000_000){
        break
    }
    if (x % 2 == 0){
        sum += x
    }
    i++
}   
console.log(sum)

File: problem-2/main.rs
fn main(){
    let mut sum = 0;
    let mut i = 1;
    loop {
        if fib(i) > 4_000_000 {
            break;
        } else {
            if fib(i) % 2 == 0 {
                sum += fib(i);
            }
            i += 1;
        }
    }
    println!("{}",sum);
}

fn fib(n: u128) -> u128 {
    match n {
        0 => 1,
        1 => 1,
        _ => fib(n - 1) + fib(n - 2),
    }
}